### PR TITLE
(SIMP-10262) Pin 1 component and update assets

### PR DESCRIPTION
**Pull request by lnemsick-simp**
Updates:
- Pin simp-nfs version to 7.1.0
- Ensure removed RPMs are obsoleted in simp-extras, or upgrades
  will fail because of dependency issues
- Remove OBE reference to puppetlabs-mount_providers in dependencies.yaml
- Update static assets (minor)
- Add location current location of dev key directory to pkg:build_clean

SIMP-10262 #comment part 4

### CODE REVIEW

**Review by jeannegreulich — CHANGES_REQUESTED** [review]
x2go and mate still work with El7.  We shouldn't need to remove them completely.  And the deps should remain in in package.yaml for el7.

File: src/assets/simp/build/simp.spec [hunk 1]
```diff
@@ -133,6 +133,14 @@ Obsoletes:  pupmod-puppetlabs-docker  <= 3.5.0
 Obsoletes:  pupmod-simp-simp_docker  <= 0.2.1
 Obsoletes:  pupmod-simp-journald   <= 1.1.0
 
+#  These lines are required for upgrades from 6.5.0 to 6.6.0
```
Comment (trevor-vaughan) [inline src/assets/simp/build/simp.spec:136]: We shouldn't need to do any of this past 6.4.0.

Packages are now inert by default so, while they may be dangling, there's no reason to make a mess of `Obsoletes` that we may have to redact in the future.

I may be missing something, could you point at an example where this causes issues?

**Comment by trevor-vaughan** [comment]
I'm up in the air on this. While I'm OK with the change (it gets rid of a LOT of chaff and doesn't break existing systems), I also agree with the point by @jeannegreulich that there's no reason to not provide something that works for EL7 systems regardless of the server managing the target systems.

That said, EL7 if all but dead at this point so we might just want to jettison anything that's not going to be maintained moving forward to lighten the maintenance burden.

**Comment by op-ct** [comment]
It's probably too soon to drop the modules—they still work, we didn't announce any deprecations for them in earlier releases, and each modules' `metadata.json` file only claims to support EL7.

**Review by jeannegreulich — APPROVED** [review]

**Pull request merged**